src/vibrator-hidl.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 UBports foundation
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Marius Gripsgard <marius@ubports.com>
+ * Author: Erfan Abdi <erfangplus@gmail.com>
+ */
+
+#include "vibrator-hidl.h"
+
+#include <iostream>
+#include <unistd.h>
+
+namespace hfd {
+
+bool VibratorHIDL::usable() {
+    return u_hardware_vibrator_new() != NULL;
```

Comment:
We should avoid calling u_hardware_vibrator_new() twice and leaking its result if it succeeds during usable() check. I'm not sure of best solution, but how about having try_create() static method that would just return a smart pointer to VibratorHIDL if succeds?